### DISCUSSION TITLE
Welcome Newcomers: We Need You!

### DISCUSSION THREAD

**Discussion by xee5ch** [oscal-club/community#9] [General]
If you are here, you probably read my announcement on [the oscal-dev mailing list](https://list.nist.gov/oscal-dev) on the evening of 4 February 2021 and I definitely welcome all of you!

Take a look at the README and of course the current [Code of Conduct](https://github.com/oscal-club/community/blob/c0769d9ddd24ffc756c37e023e162bcd05bfd24c/CODE_OF_CONDUCT.md).

This is a call to arms: we need to band together and say what things in the ecosystem, outside of just OSCAL, we need. Discuss use cases, tell us about what kind of "user" of OSCAL you will be. [You might look at some existing ideas NIST has for different personas](https://pages.nist.gov/OSCAL/learnmore/stakeholders/), but I would like to hear more.

If you have questions about community tooling that go beyond what OSCAL or tool-makers can support, especially integrating the former and the latter, bring it here! That's not really their "job" and as a community we need to show each other how it's done. :smile: 

I look forward to getting to know all of you.

My personal slogan for this community: World Unification == World Domination, Have a Nice Day. Let's make that happen. 

- Al

**Comment by openprivacy** [comment]
Hello Al - glad to be here. Not totally new to this as I've been working on compliance automation for a while. After creating my first 400-page SSP in 2016 by hand, I vowed never to do that again. Since then I've been able to obtain five ATOs for four federal agencies using an OpenControl-based (and now OSCAL enabled) [SSP-Toolkit](https://github.com/CivicActions/ssp-toolkit) that contains a rough set of shared, reusable components that worked for our clients.

Now empowered with a team, our goal is to support [compliance as code policy](https://medium.com/civicactions/policy-recommendations-for-improving-the-ato-process-through-compliance-as-code-524e3005fceb) and develop a shareable component library. The team is currently building tools to:

- Refactor existing SSP control implementation statements into known reusable components
- Aid in the process of discovering and templating new reusable components
- Support the process of composing a selection of components into a verifiable SSP
- Support collaboration with other teams and tools through published data formats (e.g. OSCAL) and open APIs

Ideally, most of the distributed component library will be free and open source - minimizing the need to reinvent the wheel for deployments of similar technologies - though we expect that some agencies may also maintain private component repositories. In any case, the specific values in (say) organizational parameters may well be private.

That’s enough for an introduction. I look forward to working with you on these and related ongoing authorization projects so that - one day - being compliant will also ensure a greater level of security.

**Reply by xee5ch** [reply]
Wow, so excited to have someone I respect so much here and kicking off this thread. I look forward to more conversation with you! I guess I respond to your points line by line.

> Since then I've been able to obtain five ATOs for four federal agencies using an OpenControl-based (and now OSCAL enabled) SSP-Toolkit that contains a rough set of shared, reusable components that worked for our clients.

I had been familiar with ssp-toolkit long ago, and had experimented with OpenControl, but did not know it grew OSCAL support until, well, maybe now (I was looking at it a few days ago too, skimmed the README, and left a note to experiment later). This is most wonderful news, as some have been talking about Python libraries for operating on OSCAL, and I was interested in converting OpenControl to OSCAL separately of that, and it seems to be a best of both worlds opportunity!

I see you and @woodt work on [CUB](github.com/GovReady/CUB) too. I guess I need to work towards a more comprehensive understanding of all the OpenControl tools that are outside the github.com/opencontrol organization, and how they map together and use cases.

I wonder, do people like yourself think a mindmap has value for the community?

> Now empowered with a team, our goal is to support compliance as code policy and develop a shareable component library.

I am very excited in this idea, personally and professionally. I will explore your components more thoroughly, and I want to blow up a whole thread for people to discuss, if they can be so bold, how we can bring that passion in strategy realize some examples. Can we even make an example component library in OSCAL that is more elaborate? I saw [your components in ssp-toolkit that currently exist in what seems to be OC YAML](https://github.com/CivicActions/ssp-toolkit/blob/master/components/). Am I correct?

Anyway, you can feel free to disregard here. I will likely tonight start a dedicated discussion area about realizing Day One, because, well, before the blog post I read the original policy paper and strongly believe in you all!

> Ideally, most of the distributed component library will be free and open source - minimizing the need to reinvent the wheel for deployments of similar technologies - though we expect that some agencies may also maintain private component repositories. In any case, the specific values in (say) organizational parameters may well be private.

I think we are definitely in the same page, and I hope we can bring passionate support and neutral examples without sensitive organizational data. Our communities depend on us too! :-)

> That’s enough for an introduction. I look forward to working with you on these and related ongoing authorization projects so that - one day - being compliant will also ensure a greater level of security.

Absolutely! It might not seem emphatic, but this first post coming from someone like you, @openprivacy, is inspiring and motivating. Let's make it happen.

I am workshopping a slogan I have used at work, and I feel maybe will become the motto and core principle of this group. If someone like you will come to such a nascent community, I feel it is time!

> World Unification == World Domination. Have a Nice Day!

**Comment by openprivacy** [comment]
Great to get such a complete response from the host of this club. :100:

Love to hear the ssp-toolkit has been reviewed, but the components in it (source in OpenControl, converted with compliance-io from @woodt) are rough at best, though they worked as a proof of concept.

I believe the component library - that I expect will be in OSCAL - will/should be distributed and based in git. The tools (lightweight and composable) will be able to pull from any library they have access to. I have some ideas on the library structure that I will share as soon as I get them down on virtual paper.

Nice to make the e-acquaintance. WUWD.HND!

**Reply by xee5ch** [reply]
I _love_ you already picked up on the slogan. You made me smile as I wrap up my day!

**Comment by mruge** [comment]
@openprivacy I am glad to hear you have had success getting ATOs.  Most the customers I have worked, we were lucky enough to get them to accept Documentation as Code Gitbook documentation instead of XLS and DOC.

I looked into the some of the SSP toolkit and compliance masonry tools when I first started looking into OSCAL (recently).  My main objection was it was using Gitbook and had limited customization for the output.    

Ill take this opportunity to give a brief rundown of what I have started working on, in case we can find some way to collaborate and make something more complete and useful to others.

* **PYOSCAL Library**:  parses oscal XML files (reverse engineered from the example repos.)  And stores them in python custom objects.
* **Sphinx-oscal**: Reads a list of oscal source (git repos), clones them, and creates OSCAL objects for use in other plugins.  mainly Jinja and Plotly for rendering the content from them.
* **Flask**: create some apps for authoring and editing...aided by the pyoscal tool these are barely POCs at best.

I chose this method rather than the compliance masonry, mostly for the flexibility, and compatibility with sphinx (I prefer it over gitbook).

There are far better ways of making these modules (meta schemas for example).  And with a bit of work, i think they could be easily adapted for better front end tools.

PS.  Very happy to find these communities, I really want to join this fight.  I have a rare and special place in my heart for documentation as code.

**Reply by xee5ch** [reply]
@mruge, there is a lot to unpack here, but I am willing to set you up with a repo and full admin rights if we want a place to have community maintenance on these different things, one or two or three. I want to start up an essentials blog, more about approachability, but then highlight more tangible examples that show how (with code and apps from both of you) _after_ we explain the why, especially to those who are less technical and more managerial.

Since it might be work product you cannot share in total, that's fine too. You can host or maybe I fork examples here, but if you are not comfortable hosting 100% here, I would love to keep you and Fen in the lineup of wonderful examples of the community. :-)

**Reply by openprivacy** [reply]
@mruge glad to hear from you. I suspect there are a number of us looking for a better way to handle SSPs.

To be clear, I based the ATOs on the ssp-toolkit, but to this date I have not gotten an AO to look at a git repo - I used the `exportto` operation and `pandoc` to generate Word files consumed by CSAM or eMASS.

@xee5ch not sure if we need a mind map, but maybe an index with pointers to the various FOSS libraries - and a few words about what each can do - would be useful.
